django-stubs/db/models/fields/related.pyi

```diff
@@ -124,6 +134,7 @@ class ForeignObject(RelatedField[_ST, _GT]):
         error_messages: _ErrorMessagesMapping | None = ...,
         db_comment: str | None = ...,
     ) -> None: ...
+    def __copy__(self) -> Empty: ...
```

Comment:
I think this should actually be `Self`, the `Empty` class is just an artifact of a weird hack that Django uses, but it modifies the `__class__` attribute so that it's no longer actually the `Empty` class

```python
    def __copy__(self):
        # We need to avoid hitting __reduce__, so define this
        # slightly weird copy construct.
        obj = Empty()
        obj.__class__ = self.__class__
        obj.__dict__ = self.__dict__.copy()
        return obj
```

Reply:
Based on your explanation, that seems correct. Thank you for pointing that out!